src/bmp/BMPParser.cc

```diff
@@ -157,39 +156,42 @@ void Parser::parse(uint8_t *buf, int bufSize, uint8_t *format){
     skip(8);
 
     // Number of colors in the palette or 0 if no palette is present
-    auto palColNum = U4();
-    EU(palColNum, "non-empty color palette");
+    palColNum = U4();
 
-    // Number of important colors used or 0 if all colors are important
-    auto impCols = U4();
-    EU(impCols, "non-zero important colors");
+    // Number of important colors used or 0 if all colors are important (generally ignored)
+    skip(4);
 
-    // BITMAPV4HEADER has additional properties
-    if(dibSize == 108){
+    if(infoHeader >= 2){

```

Comment:
With this change, the header might not have an alpha mask (V2), right? So I think `alpha` needs to be parsed/skipped conditionally in the `if/else` below?

Reply:
Done